### ISSUE TITLE
HTML generation runs out of memory

### ISSUE THREAD

**Issue by Synchro** [kalessil/phpinspectionsea#1128]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Bug                          |
| Plugin         | Php Inspections (EA Extended) 3.0.8.1 |
| Language level | PHP 7.1                     |

Current behaviour
---
I have done an inspection (of a medium sized Laravel project) and got the results, and I want to export it as HTML. When I select that option, it shows a modal progress dialogue, which doesn't do anything. After about 10 minutes, I get an error from PHPStorm saying it's run out of memory - my current allocation is 8Gb, which seems a little excessive for a bit of HTML!

In contrast, the XML exporter works perfectly on the same set of inspection results and only takes a few seconds to run.

Expected behaviour
---
It generates HTML inspection results without running out of memory.

Environment details
---
PhpStorm 2018.3.1
Build #PS-183.4588.24, built on November 28, 2018
JRE: 1.8.0_152-release-1343-b16 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Mac OS X 10.11.6

**Comment by funivan** [comment]
1. How many errors do you have?
2. Do you try to do the same operation with disabled plugin PhpInspeactions EA?

**Comment by Synchro** [comment]
You're right - I don't think this is a PIEA problem, it's more about PHPStorm. I'm not sure why I didn't think of that before, sorry! In this case I think it was due to two CSV files in the repo causing > 20k errors. I removed them from the inspection profile and though it still took a while (and generated a 155M HTML file), it did eventually finish. I would report it to JetBrains but their bug tracker appears to be down.
